### PR TITLE
Added support for returning ranked issue idxs

### PR DESCRIPTION
**Pull request by ulya-tkch**
- code
- tests
- docstring

### CODE REVIEW

File: cleanlab/rank.py [hunk 1]
```diff
@@ -553,3 +553,29 @@ def get_confidence_weighted_entropy_for_each_label(
     label_quality_scores = np.log(label_quality_scores + 1) / clipped_scores
 
     return label_quality_scores
+
+
+def find_top_issues(quality_scores, top=20):
```
Comment (jwmueller) [inline cleanlab/rank.py:558]: add types ?

File: cleanlab/rank.py [hunk 2]
```diff
@@ -553,3 +553,29 @@ def get_confidence_weighted_entropy_for_each_label(
...
+      Scores array of shape ``(N,)``, where N is the number of examples.
+
+    top : int, optional
+      The number of top scores to return.
+
+    Returns
+    -------
+    top_issue_indices : np.ndarray
+      Top indices of issue examples ranked by how likely they are to be an issue.
+    """
```
Comment (elisno) [inline cleanlab/rank.py:575]:
```suggestion
def find_top_issues(quality_scores: np.ndarray, top:  Optional[int]=20) -> np.ndarray:
    """
    Uses passed in `quality_scores` to return `top` indices corresponding to examples most likely to be issues.
    Indices ordered from most to least likely to be an issue based on value of quality_score.

    Parameters
    ----------
    quality_scores : 
      Scores array of shape ``(N,)``, where N is the number of examples.

    top :
      The number of top scores to return.

    Returns
    -------
    top_issue_indices :
      Top indices of issue examples ranked by how likely they are to be an issue.
    """
```

**Review by jwmueller — APPROVED** [review]
LGTM!

File: cleanlab/rank.py [hunk 3]
```diff
@@ -553,3 +553,29 @@ def get_confidence_weighted_entropy_for_each_label(
     label_quality_scores = np.log(label_quality_scores + 1) / clipped_scores
 
     return label_quality_scores
+
+
+def find_top_issues(quality_scores: np.ndarray, top: int = 20) -> np.ndarray:
+    """
+    Uses passed in `quality_scores` to return `top` indices corresponding to examples most likely to be issues.
+    Indices ordered from most to least likely to be an issue based on value of quality_score.
```
Comment (jwmueller) [inline cleanlab/rank.py:561]:
```suggestion
    Uses provided `quality_scores` to return `top` indices corresponding to the examples most likely to exhibit issues of a particular type.
    Returned indices are ordered from most to least severe instance of issue (based on value of `quality_score`). 
    Make sure that the ordering of `quality_scores` matches the examples in your dataset.
```

File: cleanlab/rank.py [hunk 4]
```diff
@@ -553,3 +553,29 @@ def get_confidence_weighted_entropy_for_each_label(
...
+
+    Parameters
+    ----------
+    quality_scores :
+      Scores array of shape ``(N,)``, where N is the number of examples.
```
Comment (jwmueller) [inline cleanlab/rank.py:566]:
```suggestion
      Array of shape ``(N,)``, where N is the number of examples, containing one quality score for each example in the dataset.
```

File: cleanlab/rank.py [hunk 5]
```diff
@@ -553,3 +553,29 @@ def get_confidence_weighted_entropy_for_each_label(
...
+
+    top :
+      The number of top scores to return.
```
Comment (jwmueller) [inline cleanlab/rank.py:569]:
```suggestion
      The number of indices to return.
```

File: cleanlab/rank.py [hunk 6]
```diff
@@ -553,3 +553,29 @@ def get_confidence_weighted_entropy_for_each_label(
...
+
+    Returns
+    -------
+    top_issue_indices :
+      Top indices of issue examples ranked by how likely they are to be an issue.
```
Comment (jwmueller) [inline cleanlab/rank.py:574]:
```suggestion
      Indices of top examples most likely to suffer from issue (ranked by issue severity). 
```

File: cleanlab/outlier.py [hunk 7]
```diff
@@ -38,6 +38,8 @@ class OutOfDistribution:
     Each example's OOD score lies in [0,1] with smaller values indicating examples that are less typical under the data distribution.
     OOD scores may be estimated from either: numeric feature embeddings or predicted probabilities from a trained classifier.
 
+    To get indices of examples that are most likely outliers call :py:func:`find_top_issues <cleanlab.rank.find_top_issues>` function on computed OOD scores.
```
Comment (jwmueller) [inline cleanlab/outlier.py:41]:
```suggestion
    To get indices of examples that are the most severe outliers, call :py:func:`find_top_issues <cleanlab.rank.find_top_issues>` method on the returned OOD scores.
```

File: cleanlab/rank.py [hunk 8]
```diff
@@ -557,21 +557,22 @@ def get_confidence_weighted_entropy_for_each_label(
...
 
     top :
-      The number of top scores to return.
+      The number of indices to return.
 
     Returns
     -------
     top_issue_indices :
-      Top indices of issue examples ranked by how likely they are to be an issue.
+      Indices of top examples most likely to suffer from issue (ranked by issue severity).
```
Comment (elisno) [inline cleanlab/rank.py:575]:
```suggestion
      Indices of top examples most likely to suffer from an issue (ranked by issue severity). 
```

**Review by elisno — CHANGES_REQUESTED** [review]

File: tests/test_rank.py [hunk 9]
```diff
@@ -378,3 +380,47 @@ def test_unsupported_method_for_adjust_pred_probs():
...
+    X_test_with_ood = np.vstack([X_test, X_ood])  # Add OOD datapoint to X_test
+
+    # Create OOD object (use knn without cosine metric to identify X_ood correctly)
+    knn = NearestNeighbors(n_neighbors=5).fit(X_train)
+    ood_outlier = outlier.OutOfDistribution(params={"knn": knn})
+    ood_scores = ood_outlier.score(features=X_test_with_ood)
+
+    # Get top ood score for outlier example
+    top_outlier_indices = rank.find_top_issues(quality_scores=ood_scores, top=len(ood_scores))
+    top_outlier_indices_more_k = rank.find_top_issues(quality_scores=ood_scores, top=100000)
```
Comment (jwmueller) [inline tests/test_rank.py:398]:
```suggestion
    top_outlier_indices_more_k = rank.find_top_issues(ood_scores, top=100000)
```

Comment (jwmueller) [reply tests/test_rank.py:398]: maybe also good to sometimes test positional args are truly positional as I suggested above, now that we commonly use `,*,` in the function signatures

File: cleanlab/rank.py [hunk 10]
```diff
@@ -553,3 +553,30 @@ def get_confidence_weighted_entropy_for_each_label(
     label_quality_scores = np.log(label_quality_scores + 1) / clipped_scores
 
     return label_quality_scores
+
+
+def find_top_issues(quality_scores: np.ndarray, top: int = 20) -> np.ndarray:
```
Comment (jwmueller) [inline cleanlab/rank.py:558]:
```suggestion
def find_top_issues(quality_scores: np.ndarray, *, top: int = 20) -> np.ndarray:
```

Comment (jwmueller) [reply cleanlab/rank.py:558]: this suggestion would make it easier to update signature in the future without breaking API

**Review by cgnorthcutt — COMMENTED** [review]
change default top=10 from 20. clean up tests

**Review by jwmueller — APPROVED** [review]
LGTM! Will merge after CI queue frees up as we first want to see if docs have been fixed but there are too many Actions Workflows atm

**Pull request merged**